src/Raven.Server/Smuggler/Documents/DatabaseSource.cs

```diff
@@ -248,6 +252,18 @@ public IDisposable GetCompareExchangeValues(out IEnumerable<(string key, long in
             }
         }
 
+        public IDisposable GetPendingClusterTransactions(out IEnumerable<ClusterTransactionCommand.SingleClusterDatabaseCommand> transaction)
+        {
+            using (var scope = new DisposableScope())
+            {
+                scope.EnsureDispose(_database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context));
+                scope.EnsureDispose(context.OpenReadTransaction());
+                // if we export the transaction without having the document, we will not check if the transaction was already applied.
```

Comment:
What is the meaning here?

Reply:
When we export transaction _with_ documents we will check whether the transaction was already executed and if so, we will skip exporting it.
If we export _without_ the documents we will always take the stored transaction. (Not sure if there is a real case when we want to do something like this, but it is necessary for the tests)